src/crypto/mod.rs

```diff
@@ -3,32 +3,31 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 use crate::error::*;
+use std::any::Any;
 
-pub trait RemotePublicKey {
-    /// Import the key component in the
-    /// binary uncompressed point representation.
-    fn from_raw(raw: &[u8]) -> Result<Self>
-    where
-        Self: Sized;
+pub(crate) mod holder;
+#[cfg(feature = "backend-openssl")]
+mod openssl;
+
+#[cfg(not(feature = "backend-openssl"))]
+pub use holder::{set_boxed_cryptographer, set_cryptographer};
+
+pub trait RemotePublicKey: Send + Sync + 'static {
     /// Export the key component in the
     /// binary uncompressed point representation.
     fn as_raw(&self) -> Result<Vec<u8>>;
+    /// For downcasting purposes.
+    fn as_any(&self) -> &Any;
```

Comment:
`Cryptographer::compute_ecdh_secret` takes `RemotePublicKey` and `LocalKeyPair`, but might need to downcast them to use specific implemented methods. So that's why I came up with this.

Reply:
See https://github.com/mozilla/rust-ece/pull/36#discussion_r313173586.